xprof/btx_timeline.cpp

```diff
@@ -72,78 +79,146 @@ static perfetto_uuid_t get_parent_counter_track_uuid(timeline_dispatch_t *dispat
   auto *process = track_descriptor->mutable_process();
   process->set_pid(hp_uuid);
   std::ostringstream oss;
-  oss << "Hostname " << hostname << " | Process " << process_id << " | Device " << did;
+  oss << "Hostname " << hostname << " | Device " << deviceIdx;
   // oss << " | " << track_name << " | uuid ";
   process->set_process_name(oss.str());
   return hp_uuid;
 }
 
-static perfetto_uuid_t
-get_counter_track_uuuid(timeline_dispatch_t *dispatch,
-                        std::unordered_map<hp_ddomain_t, perfetto_uuid_t> &counter_tracks,
-                        const std::string track_name, std::string hostname, uint64_t process_id,
-                        thapi_device_id did, thapi_domain_idx domain, int64_t unit_multiplier = 1) {
+static perfetto_uuid_t get_counter_track_uuuid(
+    timeline_dispatch_t *dispatch,
+    std::unordered_map<hp_ddomain_t, perfetto_uuid_t> &counter_tracks,
+    const std::string &track_name, const std::string &hostname, uint64_t process_id,
+    thapi_device_id did, uint32_t deviceIdx, uint64_t tHandle, thapi_domain_idx domain,
+    std::optional<Details> details = std::nullopt,
+    std::unordered_map<hp_dfsdev_t, perfetto_uuid_t> *counter_tracks_fp = nullptr) {
   perfetto_uuid_t hp_dev_uuid = 0;
-  auto [it, inserted] = counter_tracks.insert({{hostname, process_id, did, domain}, hp_dev_uuid});
-  auto &potential_uuid = it->second;
-  // Exists
-  if (!inserted)
-    return potential_uuid;
-
-  perfetto_uuid_t hp_uuid = get_parent_counter_track_uuid(dispatch, hostname, process_id, did);
-  hp_dev_uuid = gen_perfetto_uuid();
-  potential_uuid = hp_dev_uuid;
+  perfetto_uuid_t hp_uuid;
+
+  if (details && counter_tracks_fp != nullptr) {
+    auto [it, inserted] = counter_tracks_fp->insert(
+        {{hostname, process_id, did, tHandle, domain, details->RxTx}, hp_dev_uuid});
+    auto &potential_uuid = it->second;
+    if (!inserted)
+      return potential_uuid;
+
+    hp_uuid = get_parent_counter_track_uuid(dispatch, hostname, process_id, did, deviceIdx);
+    hp_dev_uuid = gen_perfetto_uuid();
+    potential_uuid = hp_dev_uuid;
+  } else {
+    auto [it, inserted] =
+        counter_tracks.insert({{hostname, process_id, did, domain, tHandle}, hp_dev_uuid});
+    auto &potential_uuid = it->second;
+    if (!inserted)
+      return potential_uuid;
+
+    hp_uuid = get_parent_counter_track_uuid(dispatch, hostname, process_id, did, deviceIdx);
+    hp_dev_uuid = gen_perfetto_uuid();
+    potential_uuid = hp_dev_uuid;
```

Comment:
Look like those instruction are the same in both branch.  I think it make sense to remove it outside of the if-else